src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -8400,6 +8404,118 @@ def vectorized_udf(df: pandas.DataFrame) -> pandas.Series:  # pragma: no cover
+            copy: bool, default True
+                Always returns new objects. If copy=False and no reindexing is required then original objects are returned.
+            fill_value: scalar, default np.nan
+                Always returns new objects. If copy=False and no reindexing is required then original objects are returned.
+
+        Returns:
+            tuple of SnowflakeQueryCompilers
+            Aligned objects.
+
+        """
+        if copy is not True:
+            ErrorMessage.not_implemented(
+                "Snowpark pandas 'align' method doesn't support 'copy=False'"
+            )
+        if level is not None:
+            ErrorMessage.not_implemented(
+                "Snowpark pandas 'align' method doesn't support 'level'"
+            )
+        if fill_value is not None:
+            # TODO: SNOW-1752860
+            ErrorMessage.not_implemented(
+                "Snowpark pandas 'align' method doesn't support 'fill_value'"
+            )
+        if axis != 0:
+            # TODO: SNOW-1752856
+            ErrorMessage.not_implemented(
+                f"Snowpark pandas 'align' method doesn't support 'axis={axis}'"
+            )
+        frame = self._modin_frame
+        other_frame = other._query_compiler._modin_frame
+
+        if self.is_multiindex(axis=axis) or other._query_compiler.is_multiindex(
+            axis=axis
+        ):
+            raise NotImplementedError(
+                "Snowpark pandas doesn't support `align` with MultiIndex"
+            )
+
+        left_result, right_result = self._modin_frame, self._modin_frame
+        left_frame, right_frame = self._modin_frame, self._modin_frame
+        left_frame_data_ids, left_index_ids, right_frame_data_ids, right_index_ids = (
+            None,
+            None,
+            None,
+            None,
+        )
+        if axis == 0:
+            (
+                left_result,
+                left_frame,
+                left_frame_data_ids,
+                left_index_ids,
+            ) = align_axis_0_left(frame, other_frame, join)
+            (
+                right_result,
+                right_frame,
+                right_frame_data_ids,
+                right_index_ids,
+            ) = align_axis_0_right(frame, other_frame, join)
+
+        left_qc = SnowflakeQueryCompiler(
+            InternalFrame.create(
+                ordered_dataframe=left_frame,
+                data_column_snowflake_quoted_identifiers=left_frame_data_ids,
+                data_column_pandas_labels=frame.data_column_pandas_labels,
+                data_column_pandas_index_names=frame.data_column_pandas_index_names,
+                data_column_types=frame.cached_data_column_snowpark_pandas_types,
+                index_column_snowflake_quoted_identifiers=left_index_ids,
+                index_column_pandas_labels=left_result.index_column_pandas_labels,
+                index_column_types=left_result.cached_index_column_snowpark_pandas_types,
```

Comment:
Isn't it possible for the index column type to change, e.g. if we align two dataframes, where one has an int index, and the other a string index? IIRC, you put in a fix for this so the join won't fail on default, so the index type will be different after right?

Reply:
Yes that's a good point, this doesn't cover the case for differing index types yet. I believe I'll have to do a similar fix here, convert the index types to variant so the join won't fail?

Reply:
@sfc-gh-rdurrani I added the fix so now frames are converted to variant if the index types are not compatible